### ISSUE TITLE
Project crashes when pausing and unpausing with a NavigationAgent in the scene

### ISSUE THREAD

**Issue by stesproject** [godotengine/godot#100618]
### Tested versions

- Reproducible in 4.4dev6
- Not reprudicible in 4.4dev5 or below

### System information

Godot v4.4.dev6 - Ubuntu 24.04.1 LTS 24.04 on X11 - X11 display driver, Multi-window, 1 monitor - Vulkan (Forward+) - dedicated NVIDIA GeForce MX330 (nvidia; 550.120) - 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz (8 threads)

### Issue description

The game crashes when first setting
`get_tree().paused = false`
and then
`get_tree().paused = true`
and a NavigationAgent (2D or 3D it does not matter) is in the scene.

### Steps to reproduce

- Create a new project with Godot 4.4dev6
- Create a new scene
- Add a NavigationAgent in the scene
- Add somewhere a script that first set paused to true and then paused to false

### Minimal reproduction project (MRP)

[navigation-agent-crash.zip](https://github.com/user-attachments/files/18203195/navigation-agent-crash.zip)

**Comment by matheusmdx** [comment]
This was a regression from pr #99646 and was already fixed by pr #100105, the fix will come in the next dev release.

**Comment by akien-mga** [comment]
Indeed, and dev7 was just released, so you should be able to confirm that.
